lib/LedgerSMB/Magic.pm

```diff
@@ -0,0 +1,389 @@
+
+    # our order entry classes
+    OEC_SALES_ORDER    => 1,
+    OEC_PURCHASE_ORDER => 2,
+    OEC_QUOTATION      => 3,
+    OEC_RFQ            => 4,
+
+    # our asset report classes
+    RC_DEPRECIATION     => 1,
+    RC_DISPOSAL         => 2,
+    RC_IMPORT           => 3,
+    RC_PARTIAL_DISPOSAL => 4,
+
+    # codes
+
+    # EDI
+    EDI_PATHNAME_MAX => 180,    # default max length of EDI pathname
+                                # TODO possible fencepost error XXX
+
+    # HTTP  These can be taken from HTTP::Status
+    HTTP_OK                    => 200,  # TODO  use HTTP::Status XXX
+    HTTP_FOUND                 => 302,
+    HTTP_SEE_OTHER             => 303,
+    HTTP_BAD_REQUEST           => 400,
+    HTTP_UNAUTHORIZED          => 401,
+    HTTP_INTERNAL_SERVER_ERROR => 500,
+
+    HTTP_454 => 454,        # TODO  find the right HTTP code XXX
+
+};
+
+1;
+
+# Other magical stuff that might help
+
+#INSERT INTO lsmb_module (id, label)
+#VALUES (1, 'AR'),
+#       (2, 'AP'),
+#       (3, 'GL'),
+#       (4, 'Entity'),
+#       (5, 'Manufacturing'),
+#       (6, 'Fixed Assets'),
+#       (7, 'Timecards');
+
+#INSERT INTO location_class(id,class,authoritative)
+#VALUES ('1','Billing',TRUE);
+#       ('2','Sales',FALSE);
+#       ('3','Shipping',FALSE);
+#       ('4','Physical',TRUE);
+#       ('5','Mailing',FALSE);
+
+# INSERT INTO salutation (id,salutation) VALUES
+#    ('1','Dr.');
+#    ('2','Miss.');
+#    ('3','Mr.');
+#    ('4','Mrs.');
+#    ('5','Ms.');
+#    ('6','Sir.');
+
+# INSERT INTO contact_class (id,class) values
+#   (1,'Primary Phone');
+#   (2,'Secondary Phone');
+#   (3,'Cell Phone');
+#   (4,'AIM');
+#   (5,'Yahoo');
+#   (6,'Gtalk');
+#   (7,'MSN');
+#   (8,'IRC');
+#   (9,'Fax');
+#   (10,'Generic Jabber');
+#   (11,'Home Phone');
+#   -- The e-mail classes are hard-coded into LedgerSMB/Form.pm by class_id
+#   -- i.e. 'class_id's 12 - 17
+#   (12,'Email');
+#   (13,'CC');
+#   (14,'BCC');
+#   (15,'Billing Email');
+#   (16,'Billing CC');
+#   (17,'Billing BCC');
+#   (18,'EDI Interchange ID');
+#   (19,'EDI ID');
+
+# INSERT INTO business_unit_class (id, label, active, ordering)
+# VALUES (1, 'Department', '0', '10'),
+#       (2, 'Project', '0', '20'),
+#       (3, 'Job', '0', '30'),
+#       (4, 'Fund', '0', '40'),
+#       (5, 'Customer', '0', '50'),
+#       (6, 'Vendor', '0', '60'),
+#       (7, 'Lot',  '0', 50);
+
+# INSERT INTO jctype (id, label, description, is_service, is_timecard)
+# (1, 'time', 'Timecards for project services', true, true);
+# (2, 'materials', 'Materials for projects', false, false);
+# (3, 'overhead', 'Time/Overhead for payroll, manufacturing, etc', false, true);
+
+# menu node's 1 -- 253
+
+# Menu attribute ids  1 -- 681
+
+__END__
+
+=head1 NAME
+
+LedgerSMB::Magic - Magic number constants for LedgerSMB
+
+=head1 SYNOPSIS
+
+ use LedgerSMB::Magic qw( EC_EMPLOYEE );
+
+ if ($entity_code == EC_EMPLOYEE) {
+     print "This entity  is an employee.";
+ }
+
+=head1 DESCRIPTION
```

Comment:
Please move the POD sections for each group of constants to be immediately before each group.
That assists with consistency between the code and documentation in the long run.